pdata/internal/metrics_test.go

```diff
@@ -691,15 +691,20 @@ func TestMetricsClone(t *testing.T) {
 func TestMetricsDataPointFlags(t *testing.T) {
 	gauge := generateTestGauge()
 
-	gauge.DataPoints().At(0).SetFlags(NewMetricDataPointFlags())
+	gauge.DataPoints().At(0).SetFlags(NewEmptyMetricDataPointFlags())
 	assert.True(t, gauge.DataPoints().At(0).Flags() == MetricDataPointFlagsNone)
-	assert.False(t, gauge.DataPoints().At(0).Flags().HasFlag(MetricDataPointFlagNoRecordedValue))
+	assert.False(t, gauge.DataPoints().At(0).Flags().NoRecordedValue())
 	assert.Equal(t, "FLAG_NONE", gauge.DataPoints().At(0).Flags().String())
 
-	gauge.DataPoints().At(0).SetFlags(NewMetricDataPointFlags(MetricDataPointFlagNoRecordedValue))
+	gauge.DataPoints().At(0).SetFlags(NewEmptyMetricDataPointFlags().SetNoRecordedValue(true))
```

Comment:
gauge.DataPoints().At(0).Flags().SetNoRecordedValue(true)

Reply:
If you're meaning 

```Go
gauge.DataPoints().At(0).SetFlags(gauge.DataPoints().At(0).Flags().SetNoRecordedValue(true))
```

That would work.  But

```Go
gauge.DataPoints().At(0).Flags().SetNoRecordedValue(true)
```

wouldn't actually change the flag returned by Flags() since .SetNoRecordedValue(true) currently returns a new MetricDataPointFlags

Reply:
I think we should make `SetNoRecordedValue` to mutate and not have the `SetFlags`, similar with other "mutable" structs.

Reply:
We might need to keep `SetFlags` so that `CopyTo` works as expected.  Or we need a new `CopyTo` func on `MetricDataPointFlags` so that we can do `ms.Flags().CopyTo(dest.Flags())`

Reply:
Also, how should `SetNoRecordedValue` mutate?  Should we change `.Flags()` to return a pointer (I think it would be the only pdata field to do this) or should we actually make a new `SetFlagNoRecordedValue(bool)` function to replace `SetFlags`.  Then we could do 

```Go
gauge.DataPoints().At(0).SetFlagNoRecordedValue(true)
assert.False(t, gauge.DataPoints().At(0).Flags() == MetricDataPointFlagsNone)
assert.True(t, gauge.DataPoints().At(0).Flags().NoRecordedValue())
```

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/pull/5609

Reply:
That PR, is only to show how I think it makes more sense, and is consistent with all other "mutable" values.

Reply:
@bogdandrutu I updated MetricDataPointFlags to be a struct like your example PR.  I think the result is pretty clean and friendly to work with.  In addition to `NoRecordedValue` and `SetNoRecordedValue` I added a `None` and `Reset` so that users can interacted with the default flag.   `None` replaces the need for `gauge.DataPoints().At(0).Flags() == MetricDataPointFlagsNone`.